lib/puppet/type/openldap_database.rb

```diff
@@ -83,6 +83,26 @@ def insync?(is)
           "{SSHA}" + Base64.encode64("#{Digest::SHA1.digest("#{should}#{salt}")}#{salt}").chomp == is
         when /^\{SHA\}.+/
           "{SHA}" + Digest::SHA1.hexdigest(should) == is
+        when /^\{SSHA(256|384|512)\}.+/
```

Comment:
Doesn't it still fall in the `/^\{SSHA\}.+/` (line 80) and never fall in this case?